// tslint-to-eslint-config/src/converters/lintConfigs/rules/ruleConverters/no-document-domain.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.convertNoDocumentDomain = void 0;
const convertNoDocumentDomain = () => {
    return {
        rules: [
            {
                ruleArguments: [
                    {
                        message: "Forbidden write to document.domain.",
                        selector: 'AssignmentExpression[left.type="MemberExpression"][left.object.name="document"][left.property.name="domain"]',
                    },
                    {
                        message: "Forbidden write to document.domain.",
                        selector: 'AssignmentExpression[left.type="MemberExpression"][left.object.type="MemberExpression"][left.object.object.name="window"][left.object.property.name="document"][left.property.name="domain"]',
                    },
                ],
                ruleName: "no-restricted-syntax",
            },
        ],
    };
};
exports.convertNoDocumentDomain = convertNoDocumentDomain;
