// tslint-to-eslint-config/src/converters/lintConfigs/rules/ruleConverters/template-accessibility-label-for.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.convertTemplateAccessibilityLabelFor = void 0;
const convertTemplateAccessibilityLabelFor = (tslintRule) => {
    var _a, _b, _c, _d;
    return {
        rules: [
            {
                ...(tslintRule.ruleArguments.length !== 0 && {
                    ruleArguments: [
                        {
                            ...(((_a = tslintRule.ruleArguments[0]) === null || _a === void 0 ? void 0 : _a.controlComponents) && {
                                controlComponents: (_b = tslintRule.ruleArguments[0]) === null || _b === void 0 ? void 0 : _b.controlComponents,
                            }),
                            ...(((_c = tslintRule.ruleArguments[0]) === null || _c === void 0 ? void 0 : _c.labelComponents) && {
                                labelComponents: (_d = tslintRule.ruleArguments[0]) === null || _d === void 0 ? void 0 : _d.labelComponents.map((selector) => {
                                    var _a;
                                    return {
                                        inputs: ((_a = tslintRule.ruleArguments[0]) === null || _a === void 0 ? void 0 : _a.labelAttributes) || [],
                                        selector,
                                    };
                                }),
                            }),
                        },
                    ],
                }),
                ruleName: "@angular-eslint/template/accessibility-label-has-associated-control",
            },
        ],
        plugins: ["@angular-eslint/eslint-plugin-template"],
    };
};
exports.convertTemplateAccessibilityLabelFor = convertTemplateAccessibilityLabelFor;
