// tslint-to-eslint-config/src/converters/lintConfigs/rules/ruleConverters/use-primitive-type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.convertUsePrimitiveType = void 0;
const convertUsePrimitiveType = () => {
    return {
        rules: [
            {
                ruleName: "no-new-wrappers",
            },
            {
                ruleArguments: [
                    {
                        types: {
                            String: {
                                message: "Use string instead",
                                fixWith: "string",
                            },
                            Boolean: {
                                message: "Use boolean instead",
                                fixWith: "boolean",
                            },
                            Number: {
                                message: "Use number instead",
                                fixWith: "number",
                            },
                        },
                    },
                ],
                ruleName: "@typescript-eslint/ban-types",
            },
        ],
    };
};
exports.convertUsePrimitiveType = convertUsePrimitiveType;
